lib/senkyoshi/models/staff_info.rb

```diff
@@ -42,51 +29,68 @@ def parse_name(contact)
-      @id = xml.xpath("//STAFFINFO/@id").text
-      @title = xml.xpath("//STAFFINFO/TITLE/@value").text
-      @bio = xml.xpath("//BIOGRAPHY/TEXT").text
-      @name = parse_name(contact)
-      @email = xml.xpath("//CONTACT/EMAIL/@value").text
-      @phone = xml.xpath("//CONTACT/PHONE/@value").text
-      @office_hours = xml.xpath("//OFFICE/HOURS/@value").text
-      @office_address = xml.xpath("//OFFICE/ADDRESS/@value").text
-      @home_page = xml.xpath("//HOMEPAGE/@value").text
-      @image = xml.xpath("//IMAGE/@value").text
+      @id ||= xml.xpath("//STAFFINFO/@id").text || Senkyoshi.create_random_hex
+      @title ||= xml.xpath("//STAFFINFO/TITLE/@value").text
+      bio = xml.xpath("//BIOGRAPHY/TEXT").text
+      name = parse_name(contact)
+      email = xml.xpath("//CONTACT/EMAIL/@value").text
+      phone = xml.xpath("//CONTACT/PHONE/@value").text
+      office_hours = xml.xpath("//OFFICE/HOURS/@value").text
+      office_address = xml.xpath("//OFFICE/ADDRESS/@value").text
+      home_page = xml.xpath("//HOMEPAGE/@value").text
+      image = xml.xpath("//IMAGE/@value").text
 
-      @@entries << construct_body
+      @entries << construct_body(
+        bio: bio,
+        name: name,
+        email: email,
+        phone: phone,
+        office_hours: office_hours,
+        office_address: office_address,
+        home_page: home_page,
+        image: image,
+      )
 
       self
     end
 
-    def add_to_body(str, var)
-      @body << str if var && !var.empty?
+    def append_str(body, str, var)
+      body << str if var && !var.empty?
     end
 
-    def construct_body
-      @body = "<div>"
-      add_to_body "<img src=#{@image}/>", @image
-      add_to_body "<h3>#{@name}</h3>", @name
-      add_to_body "<p>#{@bio}</p>", @bio
+    def construct_body(opts)
+      body = "<div>"
+      append_str body, "<img src=#{opts[:image]}/>", opts[:image]
+      append_str body, "<h3>#{opts[:name]}</h3>", opts[:name]
+      append_str body, "<p>#{opts[:bio]}</p>", opts[:bio]
 
-      @body << "<ul>"
-      add_to_body "<li>Email: #{@email}</li>", @email
-      add_to_body "<li>Phone: #{@phone}</li>", @phone
-      add_to_body "<li>Office Hours: #{@office_hours}</li>", @office_hours
-      add_to_body "<li>Office Address: #{@office_address}</li>", @office_address
-      add_to_body "<li>Home Page: #{@home_page}</li>", @home_page
-      @body << "</ul></div>"
+      body << "<ul>"
+      append_str body, "<li>Email: #{opts[:email]}</li>", opts[:email]
+      append_str body, "<li>Phone: #{opts[:phone]}</li>", opts[:phone]
+      append_str(
+        body,
+        "<li>Office Hours: #{opts[:office_hours]}</li>",
+        opts[:office_hours],
+      )
+      append_str(
+        body,
+        "<li>Office Address: #{opts[:office_address]}</li>",
+        opts[:office_address],
+      )
+      append_str(
+        body,
+        "<li>Home Page: #{opts[:home_page]}</li>",
+        opts[:home_page],
+      )
```

Comment:
Lines 68-84 are doing the same thing for each option. Could we dry it up like this?

```ruby
[:email, :phone, :office_hours, :office_address, :home_page].each do |key|
  append_str body, "<li>#{sym_to_title(key)}: #{opts[key]}</li>", opts[key]
end
```

The `sym_to_title` method could look something like this.
```ruby
def sym_to_title(symbol)
  symbol.to_s.gsub("_", " ").split(/\W/).map(&:capitalize).join(" ")
end
```

I feel like that would make it easier to read as well.